pkg/rest/server.go

```diff
@@ -89,14 +112,55 @@ func StartRESTServer(port string, brokers []string) {
 			log.Printf("Error during server shutdown: %v", err)
 		}
 	}()
-	
+
 	log.Printf("REST API server running on port %s...", port)
 	if err := s.Start(); err != http.ErrServerClosed {
 		log.Fatalf("Server error: %v", err)
 	}
 }
 
+func createNewClient(w http.ResponseWriter, r *http.Request, s *Server) (status bool, err error) {
+	vars := mux.Vars(r)
+	broker := vars["broker"]
+
+	fmt.Println("Creating new Kafka client for broker:", broker)
+
+	if broker == "" {
+		http.Error(w, "Broker not specified", http.StatusBadRequest)
+		return false, fmt.Errorf("broker not specified")
+	}
+
+	config := sarama.NewConfig()
+	config.Version = sarama.V2_8_0_0
+	client, err := sarama.NewClient([]string{broker}, config)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to create Kafka client: %v", err), http.StatusInternalServerError)

```

Comment:
avoid returning the raw error message in the response. See https://owasp.org/www-community/Improper_Error_Handling for more details